### PR TITLE
withdrawal and erc20 E2E test fixes

### PR DESCRIPTION
**Pull request by siosw**
# Goals of PR

based off of @MoonShiesty's #260 and adds fixes for 
- batches including first tx with empty to field
- allows compilation of smart contracts without IR

erc20 and withdraw are passing

### CODE REVIEW

**Review by MoonShiesty — APPROVED** [review]
nice! looks good

File: services/sidecar/rollup/services/validator/state_commitment.go [hunk 1]
```diff
@@ -25,24 +26,27 @@ type VersionedStateCommitment interface {
...
+	l2StateRoot common.Hash
 }
 
 func (o *StateCommitmentV0) Version() types.Bytes32 {
 	return StateCommitmentVersionV0
 }
 
 func (o *StateCommitmentV0) Marshal() []byte {
-	var buf [64]byte
+	var buf [96]byte
```
Comment (ujvl) [inline services/sidecar/rollup/services/validator/state_commitment.go:38]: nit: define global constants to name these

File: sbin/run_e2e_tests.sh [hunk 2]
```diff
@@ -87,6 +87,14 @@ until [ -f "$ROLLUP_CFG_PATH" ]; do
   sleep 4
 done
 
+DEPLOYMENTS_ENV=".deployments.env"
```
Comment (ujvl) [inline sbin/run_e2e_tests.sh:90]: where does this come from? i dont see it in any `config` dirs

File: contracts/scripts/e2e/utils.ts [hunk 3]
```diff
@@ -169,3 +171,67 @@ export function getStorageKey(messageHash: string) {
     )
   );
 }
+
+export async function waitUntilStateRoot(l1Oracle: Contract, stateRoot: string) {
```
Comment (ujvl) [inline contracts/scripts/e2e/utils.ts:175]: nit: document

File: contracts/scripts/e2e/utils.ts [hunk 4]
```diff
@@ -169,3 +171,67 @@ export function getStorageKey(messageHash: string) {
...
+
+  let oracleStateRoot = await l1Oracle.stateRoot()
+  while (oracleStateRoot !== stateRoot) {
+    oracleStateRoot = await l1Oracle.stateRoot()
+    console.log({ stateRoot, oracleStateRoot })
+    await delay(500)
+  }
+}
+
+export async function waitUntilBlockConfirmed(rollup: Contract, blockNum: number): Promise<[number, number]> {
```
Comment (ujvl) [inline contracts/scripts/e2e/utils.ts:186]: nit: document

**Review by ujvl — APPROVED** [review]

**Pull request merged**